src/issues/stateManager.ts

```diff
@@ -80,7 +80,7 @@ export class StateManager {
 			this.context.subscriptions.push(repository.state.onDidChange(async () => {
 				if ((repository.state.HEAD ? repository.state.HEAD.commit : undefined) !== this._lastHead) {
 					this._lastHead = (repository.state.HEAD ? repository.state.HEAD.commit : undefined);
-					this.setIssueData();
+					await this.setIssueData();
 				}
 
 				const newBranch = repository.state.HEAD?.name;
```

Comment:
sorry, not directly related to this PR - but should this also be inside the check above? Seems like it would only be necessary to update this when HEAD changes

Reply:
You can have a branch change that doesn't result in the head commit being changed, so I would leave this out of the check above since it needs to happen when the branch changes, not head.